### PR TITLE
fix QByteArray::append(QString) deprecation in qt 5.15.1

### PR DESCRIPTION
**Pull request by ebbit1q**
explicitly adds conversion to qbytearray before appending of strings with toUtf8()

### CODE REVIEW

File: cockatrice/src/deckstats_interface.cpp [hunk 1]
```diff
@@ -52,7 +52,8 @@ void DeckStatsInterface::getAnalyzeRequestData(DeckList *deck, QByteArray *data)
     urlQuery.addQueryItem("deck", deckWithoutTokens.writeToString_Plain());
     urlQuery.addQueryItem("decktitle", deck->getName());
     params.setQuery(urlQuery);
-    data->append(params.query(QUrl::EncodeReserved));
+    QString queryString = params.query(QUrl::EncodeReserved);
+    data->append(queryString.toUtf8());
```
Comment (ZeldaZach) [inline cockatrice/src/deckstats_interface.cpp:56]: Can you not just say `data->append(param.query(QUrl::EncodeReserved).toUtf8());`?

Comment (ebbit1q) [reply cockatrice/src/deckstats_interface.cpp:56]: of course, it just becomes harder to read.

Comment (ZeldaZach) [reply cockatrice/src/deckstats_interface.cpp:56]: I find this a bit more confusing as-is honestly. we're setting a temp variable to just use it next line

Comment (ebbit1q) [reply cockatrice/src/deckstats_interface.cpp:56]: I disagree, it breaks up the long statement, making it easier to interpret

**Review by ZeldaZach — APPROVED** [review]

**Comment by ebbit1q** [comment]
@ZeldaZach catalina build stalled https://travis-ci.org/github/Cockatrice/Cockatrice/jobs/728014132#L85

**Comment by ZeldaZach** [comment]
Restarted

**Comment by ebbit1q** [comment]
I opted to use toUtf8 in the client but toLatin1 in the qxt code, I'm not 100% sure about which is the correct option but we might consider replacing the old qxt code with something we don't have to maintain ourselves.

**Pull request merged**